### PR TITLE
[US-3963] Visibility of properties and methods of DataFactories and Repositories is protected

### PR DESCRIPTION
**Pull request by Miroslav-Stopka**
| Q             | A
| ------------- | ---
|Description, reason for the PR| [US-3963] Visibility of properties and methods of DataFactories and Repositories is protected. Protected visibility allows overriding of behavior from projects
|New feature| No 
|BC breaks| No
|Fixes issues| ...
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/product-feed-google/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [Unreleased]
 
 ### Changed
+- visibility of all private properties and methods of DataFactories was changed to protected (@Miroslav-Stopka)
```
Comment (simara-svatopluk) [inline packages/product-feed-google/CHANGELOG.md]: Please why do You update  product-feed-* changelogs when there is nothing changed?

File: docs/wip_glassbox/wip-glassbox-customization.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # WIP Glassbox Customization
 
+## Entities
+* visibility of all private properties and methods of DataFactories was changed to protected (@Miroslav-Stopka)
```
Comment (simara-svatopluk) [inline docs/wip_glassbox/wip-glassbox-customization.md]: Please move this point down and remove Entities section. Now we have duplicated Entities section

File: docs/cookbook/adding-new-attribute-to-an-entity.md [hunk 3]
```diff
@@ -195,8 +192,6 @@ use Shopsys\FrameworkBundle\Model\Product\Unit\UnitFacade;
 
 class ProductDataFactory extends BaseProductDataFactory
```
Comment (simara-svatopluk) [inline docs/cookbook/adding-new-attribute-to-an-entity.md]: Please remove also useless uses from ProductDataFactory code. There are 3 occurrences - *Facade

File: docs/wip_glassbox/wip-glassbox-customization.md [hunk 4]
```diff
@@ -3,7 +3,10 @@
 ## Entities
 * visibility of all private properties and methods of DataFactories was changed to protected (@Miroslav-Stopka)
     * protected visibility allows overriding of behavior from projects
-    
+* visibility of all private properties and methods of repositories of entities was changed to protected (@Miroslav-Stopka)
```
Comment (simara-svatopluk) [inline docs/wip_glassbox/wip-glassbox-customization.md]: Please move this point also to the bottom Entities section

File: packages/product-feed-google/CHANGELOG.md (hunk 1 above)
Comment (PetrHeinz) [reply packages/product-feed-google/CHANGELOG.md]: https://github.com/shopsys/shopsys/pull/116/files#diff-eb626cdf8d2de8fa06580571c3ffc220

File: docs/wip_glassbox/wip-glassbox-customization.md (hunk 2 above)
Comment (Miroslav-Stopka) [reply docs/wip_glassbox/wip-glassbox-customization.md]: Done

File: docs/cookbook/adding-new-attribute-to-an-entity.md (hunk 3 above)
Comment (Miroslav-Stopka) [reply docs/cookbook/adding-new-attribute-to-an-entity.md]: Sure thing. Done.

File: docs/wip_glassbox/wip-glassbox-customization.md (hunk 4 above)
Comment (Miroslav-Stopka) [reply docs/wip_glassbox/wip-glassbox-customization.md]: Done

**Comment by Miroslav-Stopka** [comment]
Visibility of attributes and methods in ProductVisibilityRepository is changed to protected.

**Pull request merged**